test/cancelable_operation_test.dart

```diff
@@ -387,6 +387,27 @@ void main() {
         expect(runThen().value, throwsA('error'));
         originalCompleter.operation.cancel();
       });
+
+      test('after completing with a future does not invoke `onValue`', () {
+        onValue = expectAsync1((_) => '', count: 0);
+        onCancel = null;
+        var operation = runThen();
+        var workCompleter = Completer<int>();
+        originalCompleter.complete(workCompleter.future);
+        originalCompleter.operation.cancel();
+        workCompleter.complete(0);
+        expect(operation.isCanceled, true);
+      });
+
+      test('after the value is completed invokes `onValue`', () {
+        onValue = expectAsync1((_) => 'foo', count: 1);
+        onCancel = null;
```

Comment:
Why not leave it at a `count:0` expectation?

Reply:
Good call, that does add a bit more power to the test. Done.